File: lsd-interceptors/src/main/java/io/lsdconsulting/interceptors/common/Headers.java
package io.lsdconsulting.interceptors.common;

import java.util.Collection;
import java.util.Map;

import static java.util.stream.Collectors.toMap;

public class Headers {

    public enum HeaderKeys {
        SOURCE_NAME("Source-Name"),
        TARGET_NAME("Target-Name");

        private final String headerName;

        HeaderKeys(String headerName) {
            this.headerName = headerName;
        }

        public String key() {
            return headerName;
        }
    }

    public static Map<String, String> singleValueMap(Map<String, ? extends Collection<String>> headers) {
        return headers.entrySet().stream()
                .collect(toMap(Map.Entry::getKey, entry -> entry.getValue().stream()
                        .findFirst()
                        .orElse("")));
    }
}

File: lsd-interceptors/src/main/kotlin/io/lsdconsulting/interceptors/common/Headers.kt
package io.lsdconsulting.interceptors.common

fun singleValueMap(headers: Map<String, Collection<String>>): Map<String, String> {
    return headers.mapValues { it.value.firstOrNull() ?: "" }
}

enum class HeaderKeys(private val headerName: String) {
    SOURCE_NAME("Source-Name"),
    TARGET_NAME("Target-Name");

    fun key() = headerName
}
